src/Npgsql/NpgsqlTypes/PostgisTypes.cs

```diff
@@ -0,0 +1,726 @@
+            {
+                _polygons = ringList.Select(x => new PostgisPolygon(x)).ToArray();
+            }
+
+            public bool Equals(PostgisMultiPolygon other)
+            {
+                if (_polygons.Length != other._polygons.Length) return false;
+                for (int i = 0; i < _polygons.Length; i++)
+                {
+                    if (_polygons[i] != other._polygons[i]) return false;
+                }
+                return true;
+            }
+
+            public override bool Equals(object obj)
+            {
+                return obj != null && obj is PostgisMultiPolygon && Equals((PostgisMultiPolygon)obj);
+            }
+
+            public static bool operator ==(PostgisMultiPolygon x, PostgisMultiPolygon y)
+            {
+                if (object.ReferenceEquals(x, null))
+                    return object.ReferenceEquals(y, null);
+                return x.Equals(y);
+            }
+
+            public static bool operator !=(PostgisMultiPolygon x, PostgisMultiPolygon y)
+            {
+                return !(x == y);
+            }
+
+
+            public override int GetHashCode()
+            {
+                int ret = 266370105;//seed with something other than zero to make paths of all zeros hash differently.
+                for (int i = 0; i < _polygons.Length; i++)
+                {
+                    ret ^= PGUtil.RotateShift(_polygons[i].GetHashCode(), ret % sizeof(int));
+                }
+                return ret;
+            }
+
+            protected override int GetLenHelper()
+            {
+                int n = 4;
+                for (int i = 0; i < _polygons.Length; i++)
+                {
+                    n += _polygons[i].GetLen();
+                }
+                return n;
+            }
+
+
+            public int PolygonCount
+            {
+                get
+                {
+                    return _polygons.Length;
+                }
+            }
+        }
+
+        /// <summary>
+        /// Represents a collection of Postgis feature.
+        /// </summary>
+        public class PostgisGeometryCollection : IGeometry, IEquatable<PostgisGeometryCollection>, IEnumerable<IGeometry>
+        {
+            private IGeometry[] _geometries;
+
+            public IGeometry this[Int32 index]
+            {
+                get { return _geometries[index]; }
+            }
+
+            internal override OgrIdentifier Identifier
+            {
+                get { return OgrIdentifier.GeometryCollection; }
+            }
+
+            public IEnumerator<IGeometry> GetEnumerator()
+            {
+                return ((IEnumerable<IGeometry>)_geometries).GetEnumerator();
+            }
+
+            System.Collections.IEnumerator System.Collections.IEnumerable.GetEnumerator()
+            {
+                return this.GetEnumerator();
+            }
+
+            public PostgisGeometryCollection(IEnumerable<IGeometry> geometries)
```

Comment:
Wouldn't it be better to accept an array here, therefore not forcing a copy of the data structure on the user?